contracts/strategies/allocation/QFOffchainCalc.sol

```diff
@@ -0,0 +1,194 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.19;
+
+import "@solady/auth/Ownable.sol";
+
+import {IAllocationStrategy} from "../../interfaces/IAllocationStrategy.sol";
+import {Allo} from "../../core/Allo.sol";
+import {Metadata} from "../../core/libraries/Metadata.sol";
+
+/// @title QFOffchainCalc
+/// @notice A strategy that allows users to apply to a pool and vote on applications
+/// @dev This strategy is used for QF pools that have offchain calculations
+/// @author allo-team
+contract QFOffchainCalc is IAllocationStrategy, Ownable {
+    uint256 public applicationStartTime;
+    uint256 public applicationEndTime;
+    uint256 public votingStartTime;
+    uint256 public votingEndTime;
+
+    bool public identityRequired;
+
+    uint256 private _index;
+
+    Allo allo;
+
+    error APPLICATIONS_NOT_OPEN();
+    error IDENTITY_REQUIRED();
+    error VOTING_NOT_OPEN();
+
+    event ApplicationCreated(address indexed applicant, uint256 indexed applicationId);
+    event ApplicationStatusUpdated(address indexed applicant, uint256 indexed applicationId, Status status);
+    event ApplicationVoted(
+        address indexed voter, uint256 indexed applicationId, uint256 vote, address token, address recipient
+    );
+
+    /// @notice Struct to hold details of an application
+    struct Application {
+        address recipient;
+        uint256 applicationId;
+        uint256 amount;
+        uint256 applicationTime;
+        uint256 votingTime;
+        uint256 votes;
+        Status status;
+    }
+
+    // Note: this is mapped to the Allo global status's in the mapping below.
+    /// @notice Enum for the local status of the application
+    enum Status {
+        PENDING,
+        ACCEPTED,
+        REJECTED,
+        REAPPLIED
+    }
+
+    /// @notice recipient -> Status -> ApplicationStatus
+    mapping(address => mapping(Status => ApplicationStatus)) public globalStatusMap;
+
+    /// @notice recipient -> Application
+    mapping(address => Application) public applications;
+
```

Comment:
we can't use `recipient -> XYZ` mappings here, since different projects/applications can have the same receipient.
maybe we can use `applicationId` instead

Reply:
We are not using the status bitmap anymore?

Reply:
I don't think we should use status bitmap